CLC-5066 - Misc bug fixes

redconfetti opened the pull request:
https://jira.ets.berkeley.edu/jira/browse/CLC-5066

johncrossman commented:
@redconfetti and I reviewed this together. What came up:
1. why must background id be created outside of job instance?
2. is ng-show used properly? Is there an accessibility concern?
3. code styles are not standard (e.g., no parens when single arg)
4. function naming should align with biz rules

Jason has agreed to investigate #1 and talk to Christian about accessibility concern. Other items will be fixed as we go.

redconfetti commented:
I added some logger statements in to recall what happened before when I would call #background_job_initialize from within the Canvas::Egrades#canvas_course_student_grades method.

These were the calls from the Controller method:
```ruby
    egrades_worker = Canvas::Egrades.new(:canvas_course_id => canvas_course_id, :enable_grading_scheme => !!params['enableGradingScheme'])
    # egrades_worker.background_job_initialize
    egrades_worker.background.canvas_course_student_grades(true)
    Rails.logger.warn("Post Background Job Call background_job_id: #{egrades_worker.background_job_id}")
    Rails.logger.warn("Post Background Job Call background_job_report: #{egrades_worker.background_job_report}")
```

I added the call back in the Canvas::Egrades#canvas_course_student_grades method that is run as a background job like so:

```ruby
    def canvas_course_student_grades(force = false)
      background_job_initialize
      Rails.logger.warn("Background Job background_job_id: #{background_job_id}")
      course_settings = Canvas::CourseSettings.new(:course_id => @canvas_course_id)
      # ...
    end
```

This is what I got in the log:
```
13:48:00,681 INFO  [stdout] (http-localhost/127.0.0.1:3000-1) [2015-04-16 13:48:00] [WARN] Post Background Job Call background_job_id:
13:48:00] [WARN] Background Job background_job_id: Canvas::Egrades.1429217280717-27a16ca525f2d518
```

So what's happening is that the background job might not be called, and thus the #background_job_initialize isn't called to set the background_job_id, and thus the controller returns no background_job_id because it hasn't been set. So it's essentially a race condition between the call to retrieve the background_job_id and return it to the front-end, and the background job itself initializing the background job state (which includes setting a unique background job ID).

redconfetti commented:
@johncrossman So it turns out that anything inside of an element inserts into the DOM via an [ng-if](https://docs.angularjs.org/api/ng/directive/ngIf) exists within a child scope. 

See [Angularjs ng-model doesn't work inside ng-if](http://stackoverflow.com/questions/18342917/angularjs-ng-model-doesnt-work-inside-ng-if)

I didn't know this, and that's why I converted the ng-if directives into ng-show directives in the template. I thought I tried to update the ng-model directive to use $parent.enableDefaultGradingScheme, but that didn't work.

```html
<input id="cc-page-course-grading-export-enable-grading-scheme" type="checkbox"
data-ng-model="$parent.enableDefaultGradingScheme" aria-label="Enable Default Letter Grades +/- Grading Scheme">
```

Since there are essentially 3 nested ng-if directives at use, I'll need to use $parent.$parent.$parent.enableDefaultGradingScheme. I tested this out and it worked. I'll create another PR now to fix this.

@christian, do you know if ng-show properly disables a DOM element so that it's not seen by screen readers or text based browsers? Do they register `display:none` as the element not shown?

redconfetti commented:
Follow up PR #3616

christianv commented on src/assets/templates/canvas_embedded/course_grade_export.html:
```diff
@@ -119,37 +119,44 @@ <h1 class="cc-page-course-grade-export-header">Required adjustments before downl
+
+            <h3 class="cc-page-course-grade-export-error-grading-scheme-header">Default Grading Scheme</h3>
+
+            <p style="margin: 10px 0;">
```
@redconfetti can we please remove the inline styling here?

christianv commented on src/assets/templates/canvas_embedded/course_grade_export.html:
```diff
@@ -119,37 +119,44 @@ <h1 class="cc-page-course-grade-export-header">Required adjustments before downl
             </p>
+            <div>
+              <input id="cc-page-course-grading-export-enable-grading-scheme" type="checkbox"
+              data-ng-model="enableDefaultGradingScheme" aria-label="Enable Default Letter Grades +/- Grading Scheme">
```
@redconfetti the `aria-label` is redundant, since we already have an associated `label`, can we remove it?

redconfetti commented:
@johncrossman I've resolved the other two updates that we discussed in #3618
- update the Javascript function name from goToCourseDetails to goToCourseSettings
- Add more verbose comments regarding the need to fake 30 total steps before passing the background job processing to the Canvas::CourseUsers#course_users method to retrieve grades
